server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -706,6 +700,15 @@ protected void writeIndexGen(final RepositoryData repositoryData, final long rep
         }
         logger.debug("Repository [{}] updating index.latest with generation [{}]", metadata.name(), newGen);
         writeAtomic(INDEX_LATEST_BLOB, genBytes, false);
+        // delete the N-2 index file if it exists, keep the previous one around as a backup
```

Comment:
Previously we were keeping the old index-N file as a backup because a subsequent index-N write might have failed and we're left with no index files. Now we perform the cleanup latter, so is there any point to keep index-(N-1) file?

Reply:
> Previously we were keeping the old index-N file as a backup because a subsequent index-N write might have failed and we're left with no index files.

Not sure that's true. Prior to this change we are also writing the index-N first this is not changing here. I think there's a point to me made, that keeping the old index-(N-1) around protects you against broken index.latest when using the repo as the basis for a URL repository (since we write the index-N first on delete and last on snapshots this is a stable approach). Also, it's nice to maybe keep at least two of these files around for debugging so you can see the last step taken in the repository from their difference.

Reply:
Sorry, my bad. Actually, you're still writing index-N file before removing index-(N-2) file. 
Can you elaborate a bit more on this ideally with an example
>that keeping the old index-(N-1) around protects you against broken index.latest when using the repo as the basis for a URL repository (since we write the index-N first on delete and last on snapshots this is a stable approach)